fault-proof/src/lib.rs

```diff
@@ -240,3 +241,15 @@ where
 
     Ok(parent_game_contract.status().call().await? == GameStatus::CHALLENGER_WINS)
 }
+
+/// Extension trait for checking transaction error types.
+pub trait TxErrorExt {
```

Comment:
The current implementation seems very fragile. Wdyt of adding a custom error type e.g., `pub enum TxError`?